From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: conda-forge <conda-forge@conda-forge.org>
Date: Mon, 1 Jan 2025 00:00:00 +0000
Subject: [PATCH] Fix implicit int-to-float conversion in TextToSpeech_macx.mm

This patch fixes a compiler warning/error about implicit conversion from
int to float in the TextToSpeech_macx.mm file when setting the volume.

The warning occurs because 'volume' is an int parameter, and when dividing
by 100.0f, the compiler warns about the implicit conversion. The fix is to
explicitly cast the volume parameter to float.

Error:
    TextToSpeech_macx.mm:138:48: error: implicit conversion from 'int' to 'float' may lose precision [-Werror,-Wimplicit-int-float-conversion]
      138 |                 [[m_synthesizerHelper synthesizer] setVolume:volume / 100.0f];
          |                                                              ^~~~~~ ~

--- a/src/mumble/TextToSpeech_macx.mm
+++ b/src/mumble/TextToSpeech_macx.mm
@@ -135,7 +135,7 @@ void TextToSpeechPrivate::setVolume(int volume) {
 	// Check for setVolume: availability. It's only available on 10.5+.
 	if ([[m_synthesizerHelper synthesizer] respondsToSelector:@selector(setVolume:)]) {
-		[[m_synthesizerHelper synthesizer] setVolume:volume / 100.0f];
+		[[m_synthesizerHelper synthesizer] setVolume:(float)volume / 100.0f];
 	}
 }
 
